src/csharp/Grpc.Core/Marshaller.cs

```diff
@@ -57,10 +59,10 @@ public Marshaller(Action<T, SerializationContext> serializer, Func<Deserializati
         {
             this.contextualSerializer = GrpcPreconditions.CheckNotNull(serializer, nameof(serializer));
             this.contextualDeserializer = GrpcPreconditions.CheckNotNull(deserializer, nameof(deserializer));
-            // TODO(jtattermusch): once gRPC C# library switches to using contextual (de)serializer,
-            // emulating the simple (de)serializer will become unnecessary.
-            this.serializer = EmulateSimpleSerializer;
-            this.deserializer = EmulateSimpleDeserializer;
+            // gRPC only uses contextual serializer/deserializer internally, so emulating the legacy
+            // (de)serializer is not necessary.
+            this.serializer = (msg) => { throw new NotImplementedException(); };
```

Comment:
can we just get rid of the `serializer` and `deserializer` fields entirely, since they are no longer used now?

Reply:
We need to keep the Serializer and Deserializer properties to maintain backward compatibility.
And the other constructor is setting the serializer and deserializer fields.